modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/MuleArtifactClassLoader.java

```diff
@@ -199,7 +201,13 @@ public URL findResource(String name) {
             // We don't want class loaders in limbo
             synchronized (descriptorMappingLock) {
               if (descriptorMapping.get(matchDescriptor) == null) {
-                URLClassLoader urlClassLoader = new URLClassLoader(new URL[] {url});
+                URLClassLoader urlClassLoader;
+                if (isJavaVersionAtMost(JAVA_1_8)) {
+                  urlClassLoader =
+                      new URLClassLoader(new URL[] {url}, getSystemClassLoader(), new NonCachingURLStreamHandlerFactory());
```

Comment:
why not do the same as in https://github.com/mulesoft/mule/pull/12538/files#diff-37299a6d58b9aa54cab313774928f54ad995e77f8e9716b56a01be6f87b64b35R53, have a `getNonCachingURLStreamHandlerFactory()` method?

Reply:
Sure will do